compiler/noirc_evaluator/src/ssa.rs

```diff
@@ -150,15 +153,24 @@ pub(crate) fn optimize_into_acir(
 /// Run all SSA passes.
 fn optimize_all(builder: SsaBuilder, options: &SsaEvaluatorOptions) -> Result<Ssa, RuntimeError> {
     Ok(builder
-        .run_pass(Ssa::remove_unreachable_functions, "Removing Unreachable Functions")
+        .run_pass(Ssa::remove_unreachable_functions, "Removing Unreachable Functions (1st)")
         .run_pass(Ssa::defunctionalize, "Defunctionalization")
         .run_pass(Ssa::remove_paired_rc, "Removing Paired rc_inc & rc_decs")
+        .run_pass(
+            |ssa| {
+                if options.skip_preprocess_fns {
+                    return ssa;
+                }
+                ssa.preprocess_functions(options.inliner_aggressiveness)
+            },
+            "Preprocessing Functions",
+        )
```

Comment:
We end up with a lot of unreachable functions after this pass. These will be thrown away as part of the full inlining pass but it would be nice to embed `remove_unreachable_functions` within `preprocess_functions` so that printing the output doesn't contain a lot of unused functions.